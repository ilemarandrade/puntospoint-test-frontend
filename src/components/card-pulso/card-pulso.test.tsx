import React from 'react';
import { render, screen } from '@testing-library/react';

import CardPulso from './';

describe('CardPulso Component', () => {
  const mockData = {
    date: new Date('2023-10-01T04:00:00Z'),
    clients: 1500,
    totalSales: 300,
    totalAmount: 50000,
    cashbackAccumulated: 2000,
    invoiced: [
      { date: '2023-09-15', amount: 15000 },
      { date: '2023-09-20', amount: 35000 },
    ],
  };

  test('renders correctly with provided data', () => {
    const { container } = render(<CardPulso data={mockData} />);
    expect(container).toMatchSnapshot();
  });
});
