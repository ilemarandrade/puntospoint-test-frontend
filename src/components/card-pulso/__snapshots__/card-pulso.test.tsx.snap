// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPulso Component renders correctly with provided data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root px-5 pt-2 pb-5 w-full space-y-2 rounded-2xl min-w-64 undefined css-1j74uew-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <h3
      class="text-center font-semibold capitalize"
    >
      octubre
    </h3>
    <div
      class="flex justify-between"
      data-testid="card-pulso-client"
    >
      <p
        class="text-sm"
      >
        Clientes
      </p>
      <p
        class="text-sm"
      >
        1.500
      </p>
    </div>
    <div
      class="flex justify-between"
      data-testid="card-pulso-sale"
    >
      <p
        class="text-sm"
      >
        Ventas Totales
      </p>
      <p
        class="text-sm"
      >
        300
      </p>
    </div>
    <div
      class="flex justify-between"
      data-testid="card-pulso-amount"
    >
      <p
        class="text-sm"
      >
        Monto Total
      </p>
      <p
        class="text-sm"
      >
        $
        50.000
      </p>
    </div>
    <h3
      class="font-semibold"
    >
      Cashback
    </h3>
    <div
      class="flex justify-between"
      data-testid="card-pulso-accumulated"
    >
      <p
        class="text-sm"
      >
        Acumulado
      </p>
      <p
        class="text-sm"
      >
        $
        2.000
      </p>
    </div>
    <div
      class="flex justify-between"
      data-testid="card-pulso-invoice"
    >
      <p
        class="text-sm"
      >
        Facturado 
        2023-09-15
         
      </p>
      <p
        class="text-sm"
      >
        $
        15.000
      </p>
    </div>
    <div
      class="flex justify-between"
      data-testid="card-pulso-invoice"
    >
      <p
        class="text-sm"
      >
        Facturado 
        2023-09-20
         
      </p>
      <p
        class="text-sm"
      >
        $
        35.000
      </p>
    </div>
  </div>
</div>
`;
